src/cert.rs

```diff
@@ -147,16 +137,14 @@ fn version3(input: &mut untrusted::Reader) -> Result<(), Error> {
     )
 }
 
-pub fn certificate_serial_number(input: &mut untrusted::Reader) -> Result<(), Error> {
+pub fn lenient_certificate_serial_number(input: &mut untrusted::Reader) -> Result<(), Error> {
     // https://tools.ietf.org/html/rfc5280#section-4.1.2.2:
     // * Conforming CAs MUST NOT use serialNumber values longer than 20 octets."
     // * "The serial number MUST be a positive integer [...]"
-
-    let value = der::positive_integer(input)?;
-    if value.big_endian_without_leading_zero().len() > 20 {
-        return Err(Error::BadDER);
-    }
```

Comment:
Is removing this length check intentional? The rest of the PR motivation is about allowing negative serials.

With that said, this constraint is clearly on issuers and not verifiers.